hardware/EcoDevices.cpp

```diff
@@ -122,365 +119,354 @@ bool CEcoDevices::WriteToHardware(const char *pdata, const unsigned char length)
+	#endif
+
+	major = boost::lexical_cast<int>(m_status.version.substr(0,m_status.version.find(".")).c_str());
+	m_status.version.erase(0,m_status.version.find(".")+1);
+	minor = boost::lexical_cast<int>(m_status.version.substr(0,m_status.version.find(".")).c_str());
+	m_status.version.erase(0,m_status.version.find(".")+1);
+	release = boost::lexical_cast<int>(m_status.version.c_str());
+	m_status.version = S_xpath_string(XMLdoc.RootElement(),"/response/version/text()").c_str();
+
+	if ((major>min_major) || ((major==min_major) && (minor>min_minor)) || ((major==min_major) && (minor==min_minor) && (release>=min_release)))
+	{
+                m_status.hostname = S_xpath_string(XMLdoc.RootElement(),"/response/config_hostname/text()").c_str();
+		m_status.flow1    = i_xpath_int(XMLdoc.RootElement(),"/response/meter2/text()");
+		m_status.flow2    = i_xpath_int(XMLdoc.RootElement(),"/response/meter3/text()");
+		m_status.index1   = i_xpath_int(XMLdoc.RootElement(),"/response/count0/text()");
+		m_status.index2   = i_xpath_int(XMLdoc.RootElement(),"/response/count1/text()");
+		m_status.t1_ptec  = S_xpath_string(XMLdoc.RootElement(),"/response/T1_PTEC/text()").c_str();
+		m_status.t2_ptec  = S_xpath_string(XMLdoc.RootElement(),"/response/T2_PTEC/text()").c_str();
+		
+                // Process Counter 1
+		if ((m_status.index1 >0) && ((m_status.index1 != m_status.pindex1) || (m_status.flow1 != m_status.pflow1) \
+			|| (difftime(atime,m_status.time1) >= 300)))
 		{
-			m_teleinfo2.tariff="Tarif option Heures Creuses";
-			m_teleinfo2.HCHP = (unsigned long)(root["T2_HCHP"].asFloat());
-			m_teleinfo2.HCHC = (unsigned long)(root["T2_HCHC"].asFloat());
+			m_status.pindex1 = m_status.index1;
+			m_status.pflow1 = m_status.flow1;
+			m_status.time1 = atime;
+			SendMeterSensor(m_HwdID, 1, 255, m_status.index1/1000.0f, m_status.hostname + " Counter 1");
+			SendWaterflowSensor(m_HwdID, 2, 255, (float)m_status.flow1, m_status.hostname + " Flow counter 1");
 		}
 
-		// "EJP" subscription scheme
-		else if ((m_teleinfo2.PTEC == "HN..") || (m_teleinfo2.PTEC == "PM.."))
+		// Process Counter 2
+		if ((m_status.index2 >0) && ((m_status.index2 != m_status.pindex2) || (m_status.flow2 != m_status.pflow2) \
+			|| (difftime(atime,m_status.time2) >= 300)))
 		{
-			m_teleinfo2.tariff="Tarif option EJP";
-			m_teleinfo2.EJPHN = (unsigned long)(root["T2_EJPHN"].asFloat());
-			m_teleinfo2.EJPHPM = (unsigned long)(root["T2_EJPHPM"].asFloat());
+			m_status.pindex2 = m_status.index2;
+			m_status.pflow2 = m_status.flow2;
+			m_status.time2 = atime;
+			SendMeterSensor(m_HwdID, 3, 255, m_status.index2/1000.0f, m_status.hostname + " Counter 2");
+			SendWaterflowSensor(m_HwdID, 4, 255, (float)m_status.flow2, m_status.hostname + " Flow counter 2");
 		}
+	}
+	else
+	{
+		message = "EcoDevices firmware needs to be at least version ";
+		message = message + boost::to_string(min_major) + "." + boost::to_string(min_minor) + "." + boost::to_string(min_release);
```

Comment:
Could you remove the boost::to_string dependency
you can do this with a standard std::stringstream, see code elsewhere in domoticz
note, you have to clear the stringstream before using it again in the same string